### PR TITLE
feat: add utility to generate typings from introspection

### PR DESCRIPTION
**Pull request by realSpok**
## Definition of Done
#86bx3056z

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/cloud-customizer/src/types.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  TCollectionName,
+  TSchema,
+} from '@forestadmin/datasource-customizer';
+
+/**
+ * This agent the central object used to customize your cloud project
+ * some methods available in self-hosted, such as:
+ * @method addDataSource are currently not supported in cloud
+ */
+export type Agent<S extends TSchema = TSchema> = {
```
Comment (Thenkei) [inline packages/cloud-customizer/src/types.ts:14]: 👀

Comment (Thenkei) [reply packages/cloud-customizer/src/types.ts:14]: Why just declaring a type?

File: packages/cloud-customizer/src/services/typings-updater.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    envSecret: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+    isProduction: false,
+  };
+  const agent = createAgent(agentOptions);
+
+  // TODO actually get the introspection from api
+  const introspection = JSON.parse(
+    fs.readFileSync('/Users/nicolas/Forest/cloud-customizer/introspection-sample.json', 'utf8'),
+  );
+  agent.addDataSource(createSqlDataSource(`sqlite::memory:`, introspection));
```
Comment (Thenkei) [inline packages/cloud-customizer/src/services/typings-updater.ts:19]: Loving this <3

File: packages/cloud-customizer/src/services/typings-updater.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  const agentOptions: AgentOptions = {
+    authSecret: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+    envSecret: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+    isProduction: false,
+  };
+  const agent = createAgent(agentOptions);
+
+  // TODO actually get the introspection from api
+  const introspection = JSON.parse(
+    fs.readFileSync('/Users/nicolas/Forest/cloud-customizer/introspection-sample.json', 'utf8'),
```
[the remark this answers is not in the document]
Comment (realSpok) [reply packages/cloud-customizer/src/services/typings-updater.ts:17]: oups … let me fix that

**Review by EnkiP — APPROVED** [review]

**Review by EnkiP — APPROVED** [review]

File: packages/cloud-customizer/src/types.ts (hunk 1 above)
Comment (realSpok) [reply packages/cloud-customizer/src/types.ts]: it is outside the scope of this PR any. I moved it to this one:
https://github.com/ForestAdmin/agent-nodejs/pull/916

**Review by Scra3 — CHANGES_REQUESTED** [review]
waiting my push

**Review by Scra3 — APPROVED** [review]

**Pull request merged**